lightly/models/moco.py

```diff
@@ -58,87 +40,28 @@ class ResNetMoCo(nn.Module, _StateDictLoaderMixin):
     """
 
     def __init__(self,
-                 name: str ='resnet-18',
-                 width: float = 1.,
+                 backbone: nn.Module,
                  num_ftrs: int = 32,
                  out_dim: int = 128,
-                 num_splits: int = 8,
-                 m: float = 0.999):
-
-        self.num_ftrs = num_ftrs
-        self.out_dim = out_dim
-        self.num_splits = num_splits
-        self.m = m
+                 m: float = 0.999,
+                 batch_shuffle: bool = False):
 
-        super(ResNetMoCo, self).__init__()
-
-        self.features, self.projection_head = \
-            _get_features_and_projections(
-                ResNetGenerator(name=name, width=width, num_splits=num_splits),
-                self.num_ftrs,
-                self.out_dim,
-                num_splits
-            )
-
-        self.key_features, self.key_projection_head = \
-            _get_features_and_projections(
-                ResNetGenerator(name=name, width=width, num_splits=num_splits),
-                self.num_ftrs,
-                self.out_dim,
-                num_splits
-            )
-
-        # set key-encoder weights to query-encoder weights
-        for param_k in self.key_features.parameters():
```

Comment:
this is an example where you could use a more generic function that wraps a loop that deactivates gradient for parameters